crates/indexer/src/error_receipt_indexer.rs

```diff
@@ -0,0 +1,154 @@
+use anyhow::Result;
+use gw_common::H256;
+use gw_types::offchain::ErrorTxReceipt;
+use rust_decimal::Decimal;
+
+use crate::{
+    helper::{hex, parse_log, GwLog},
+    pool::POOL,
+};
+
+pub const MAX_RETURN_DATA: usize = 32;
+pub const MAX_ERROR_TX_RECEIPT_BLOCKS: u64 = 3;
+
+pub struct ErrorReceiptIndexer {
+    latest_block: u64,
+}
+
+impl Default for ErrorReceiptIndexer {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl ErrorReceiptIndexer {
+    pub fn new() -> Self {
+        ErrorReceiptIndexer { latest_block: 0 }
+    }
+
+    async fn insert_error_tx_receipt(receipt: ErrorTxReceipt) -> Result<()> {
+        let record = ErrorReceiptRecord::from(receipt);
+        log::debug!("error tx receipt record {:?}", record);
+
+        sqlx::query("INSERT INTO error_transactions (hash, block_number, cumulative_gas_used, gas_used, status_code, status_reason) VALUES ($1, $2, $3, $4, $5, $6)")
+            .bind(hex(record.tx_hash.as_slice())?)
+            .bind(Decimal::from(record.block_number))
+            .bind(Decimal::from(record.cumulative_gas_used))
+            .bind(Decimal::from(record.gas_used))
+            .bind(Decimal::from(record.status_code))
+            .bind(record.status_reason)
+            .execute(&*POOL)
+            .await?;
+
+        Ok(())
+    }
+
+    async fn clear_expired_block_error_receipt(block_number: u64) -> Result<()> {
+        let result = sqlx::query("DELETE FROM error_transactions WHERE block_number <= $1")
+            .bind(Decimal::from(block_number))
+            .execute(&*POOL)
+            .await?;
+
+        log::info!("delete error tx receipt {}", result.rows_affected());
+
+        Ok(())
+    }
+}
+
+impl ErrorReceiptIndexer {
+    pub async fn handle_error_receipt(&mut self, receipt: ErrorTxReceipt) {
+        if self.latest_block < receipt.block_number {
+            self.latest_block = receipt.block_number;
+
+            let expired_block = self
+                .latest_block
+                .saturating_sub(MAX_ERROR_TX_RECEIPT_BLOCKS);
+            smol::spawn(async move {
+                if let Err(err) = Self::clear_expired_block_error_receipt(expired_block).await {
+                    log::error!("clear expired block error receipt {}", err);
+                }
+            })
+            .detach();
+        }
+
+        smol::spawn(async move {
+            if let Err(err) = Self::insert_error_tx_receipt(receipt).await {
+                log::error!("insert error tx receipt {}", err);
+            }
+        })
+        .await;
```

Comment:
no need to spawn a future and await

```suggestion
        if let Err(err) = Self::insert_error_tx_receipt(receipt).await {
            log::error!("insert error tx receipt {}", err);
        }
```